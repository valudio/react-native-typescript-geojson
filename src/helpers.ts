import { Position } from 'geojson';

import { IOverlay } from './models';

const flatten = (prev: Array<IOverlay>, curr: Array<IOverlay>) =>
  prev.concat(curr);

const makePoint = (c: Position) => ({ latitude: c[1], longitude: c[0] });

const makeLine = (l: any) => l.map(makePoint);

export { flatten, makePoint, makeLine };
