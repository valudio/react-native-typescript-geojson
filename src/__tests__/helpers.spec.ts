import { flatten, makePoint, makeLine } from '../helpers';

test('makePoint should return point object', () => {
  const point = makePoint([0, 1]);

  expect(point).toMatchSnapshot({
    latitude: expect.any(Number),
    longitude: expect.any(Number)
  });
});

test('makeLine should return array point object', () => {
  const line = makeLine([[0, 1], [0, 1]]);

  line.forEach((point: any) => {
    expect(point).toMatchSnapshot({
      latitude: expect.any(Number),
      longitude: expect.any(Number)
    });
  });
});
