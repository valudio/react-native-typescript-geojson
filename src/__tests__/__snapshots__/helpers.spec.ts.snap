// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`makeLine should return array point object 1`] = `
Object {
  "latitude": Any<Number>,
  "longitude": Any<Number>,
}
`;

exports[`makeLine should return array point object 2`] = `
Object {
  "latitude": Any<Number>,
  "longitude": Any<Number>,
}
`;

exports[`makePoint should return point object 1`] = `
Object {
  "latitude": Any<Number>,
  "longitude": Any<Number>,
}
`;
